components/tidb_query/src/util.rs

```diff
@@ -5,6 +5,34 @@ use tipb::ColumnInfo;
 
 use crate::codec::datum::Datum;
 
+const MAX_RAND_VALUE: u64 = 0x3FFFFFFF;
+
+pub struct MySQLRng {
+    seed1: u32,
+    seed2: u32,
+}
+
+impl MySQLRng {
+    pub fn new() -> Self {
+        let current_time = time::get_time();
+        let nsec = i64::from(current_time.nsec);
+        Self::new_with_seed(nsec)
+    }
+
+    pub fn new_with_seed(seed: i64) -> Self {
+        let seed1 =
+            (seed.wrapping_mul(0x10001).wrapping_add(55555555)) as u32 % MAX_RAND_VALUE as u32;
```

Comment:
I think this also works:
```suggestion
            (seed.wrapping_mul(0x10001).wrapping_add(55555555)) % MAX_RAND_VALUE as u32;
```

Reply:
I just revert MAX_RAND_VALUE to u32.

Reply:
Now except our seed all the variable type is u32, it seem more make sense and the rand algorithm also work well.